sources/packages/backend/apps/api/src/route-controllers/student/_tests_/e2e/student.students.controller.getStudentApplicationSummary.e2e-spec.ts

```diff
@@ -0,0 +1,141 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  FakeStudentUsersTypes,
+  getRecentActiveProgramYear,
+  getStudentToken,
+  mockJWTUserInfo,
+  resetMockJWTUserInfo,
+} from "../../../../testHelpers";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  saveFakeApplication,
+  saveFakeStudent,
+} from "@sims/test-utils";
+import { TestingModule } from "@nestjs/testing";
+import { ApplicationStatus, ProgramYear } from "@sims/sims-db";
+
+describe("StudentStudentsController(e2e)-getStudentApplicationSummary", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+  let appModule: TestingModule;
+  let recentActiveProgramYear: ProgramYear;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource, module } =
+      await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    appModule = module;
+    recentActiveProgramYear = await getRecentActiveProgramYear(db);
+  });
+
+  beforeEach(() => {
+    resetMockJWTUserInfo(appModule);
+  });
+
+  it(`Should get student application summary when the student has an application in status ${ApplicationStatus.Submitted}.`, async () => {
+    // Arrange
+    const student = await saveFakeStudent(db.dataSource);
```

Comment:
I do not see a reason to have the student saved separately from the application. Unless there is a reason, I believe using a factory that creates the data setup as required should be enough.

```ts
// Arrange
const savedApplication = await saveFakeApplication(db.dataSource, {
  programYear: recentActiveProgramYear,
});
// Mock the user received in the token.
await mockJWTUserInfo(appModule, savedApplication.student.user);
```